Editor/Resources/EditorWindow/Pages/LandingPage.uss

```diff
@@ -0,0 +1,72 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+@import "./common.uss";
+
+:root {
+    --color-card-border: #0d0d0d;
+    --color-card-background: #2a2a2a;
+}
+
+.cards {
+    flex-direction: row;
+}
+
+.card {
+    padding: 12px;
+    border-width: 2px;
+    border-color: var(--color-card-border);
+    background-color: var(--color-card-background);
+    flex-basis: 50%;
+    margin-right: 24px;
+}
+
+.card--last {
+     margin-right: 0;
+ }
+
+.card__content {
+    flex-direction: row;
+    height: 60px
+}
+
+.card__icon {
+    height: 60px;
+    width: 60px;
+    flex: 0 0 auto;
+}
+
+.card__text {
+    font-size: var(--font-size-default);
+    -unity-text-align: middle-left;
+    flex: 1 1 auto;
+}
+
+.card__button {
+    font-size: var(--font-size-large);
+}
+
+.sample__header {
+    font-size: 24px;
+    margin-bottom: 24px;
+}
+
+.sample__content {
+    flex-direction: row;
+    font-size: 16px;
+}
+
+.sample__text {
+    align-items: flex-start;
+}
+
+.sample__image {
+    width: 96px;
+    height: 96px;
+    margin-right: 24px;
+}
+
+.sample__button {
+    margin-top: 8px;
```

Comment:
Commenting here as well since this is the main idea. I see this forums post for a reasonable workaround that `:last-child` doesn't exist in USS: https://forum.unity.com/threads/uss-dont-apply-to-last-element.943742/

The idea is for the container to have a negative padding to account for the last margin so we should still be able to create this generic idea

Reply:
I've tried multiple variations on this, but ran into issues each time. Negative padding is an invalid value.
Using negative margin on container and margin on container children works, however if one of the container children also has class `.separate` then it breaks in that instance by having the wrong amount of margin.
For it to work, you would have to wrap the container child in an empty div essentially.

How would you proceed?



example:
![image](https://github.com/aws/amazon-gamelift-plugin-unity/assets/109524503/0c49a188-ff4a-42e3-bfaf-bef5d699a6f8)
here the container for the whole page has classes `separate separate--vertical separate--large`. `title__section` has classes `separate separate--vertical separate--small`. You can see the game below the paragraph text inside of the container

Reply:
Firstly, a couple items:
* I think we should use `separator` instead of `separate` because a noun is more appropriate than a verb for class naming. My bad for that first suggestion.
* I think we should default `separator` to use `separator--vertical` since that will be the most used
* I think we should default `separator` to use `separator--small` since that I expect will be the most used _(I only foresee large being used for the top level and for horizontal components)_

### Option 1
I believe we can solve this with a bit of recursion:
```
.separator {
  flex-direction: column;
}

.separator.separator--horizontal {
  flex-direction: row;
}

.separator {
  margin-bottom: - var(--spacing-small);
  margin-right: - var(--spacing-small);

  > * {
    margin-bottom: var(--spacing-small);
    margin-right: var(--spacing-small);
  }

  > .separator {
    // small (parent) - small (child)
    margin-bottom: 0;
    margin-right: 0;
  }

  > .separator.separator--large {
    // small (parent) - large (child)
    margin-bottom: var(--spacing-small) - var(--spacing-large);
    margin-right: var(--spacing-small) - var(--spacing-large);
  }
}

.separator.separator--large {
  margin-bottom: - var(--spacing-large);
  margin-right: - var(--spacing-large);

  > * {
    margin-bottom: var(--spacing-large);
    margin-right: var(--spacing-large);
  }

  > .separator {
    // large (parent) - small (child)
    margin-bottom: var(--spacing-large) - var(--spacing-small);
    margin-right: var(--spacing-large) - var(--spacing-small);
  }

  > .separator.separator--large {
    // large (parent) - large (child)
    margin-bottom: 0;
    margin-right: 0;
  }
}
```

However, this does create n*n edge cases. This is _okay_ with only two sizes for the separator, but if we add a medium size, we'll have 9 edge cases and so on, which could add a lot of bloat and room for error _(this can be solved by separating these out of `common.uss` and writing a quick script to build this file for us when the time comes)_.

### Option 2
If we use the extra layer of divs, I'd like to separate them into context naming (aka page vs page__header) and functional naming (aka separator). Example:
```
<div class="page">
    <div class="separator separator--large">
        <div class="page__header">
            <div class="separator separator--small">
                <label class="page__title" />
                <label class="page__content" />
            </div>
        </div>
        <div class="page__section">
            <div class="separator separator--small">
                <label class="page__title--section" />
                <div class="cards">
                    <div class="separator separator--horizontal separator--large">
                        <div class="card">
                            <div class="separator separator--small">
                                <label class="card__content" />
                                <button class="card__button" />
                            </div>
                        </div>
                        <div class="card">
                            <div class="separator separator--small">
                                <label class="card__content" />
                                <button class="card__button" />
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <div class="page__section">
            <div class="separator separator--small">
                <label class="page__title--section" />
                <label class="page__content" />
                <button class="button" />
            </div>
        </div>
    </div>
</div>
...
```

This is compared to if we didn't have to have extra divs:
```
<div class="page separator separator--large">
    <div class="page__header separator separator--small">
        <label class="page__title" />
        <label class="page__content" />
    </div>
    <div class="page__section separator separator--small">
        <label class="page__title--section" />
        <div class="cards separator separator--horizontal separator--large">
            <div class="card separator separator--small">
                <label class="card__content" />
                <button class="card__button" />
            </div>
            <div class="card separator separator--small">
                <label class="card__content" />
                <button class="card__button" />
            </div>
        </div>
    </div>
    <div class="page__section separator separator--small">
        <label class="page__title--section" />
        <label class="page__content" />
        <button class="button" />
    </div>
</div>
```

This isn't too tedious, it feels kinda natural still; however, the indenting makes this a lot harder because most everything will be double the current indents. With 4-space indents this gets problematic for reading the code as it gets harder and harder to fit on screen.

One potential idea if this gets to be too much nesting is to create a separator Template which has full ownership of it's styling classes (so no longer in `common.uss`) and it knows to create a separate div layer to make sure recursive separators don't break margins.

### Conclusion
If option 1 works and I'm not mistaken with my recursive idea working, then I think we should use that as I foresee us using separators a lot and the divs will get annoying. If the recursion doesn't work the way I think it would work, we can use the divs idea. That's my take, let me know how that goes/if you have other ideas!